src/framework/components/sound/system.js

```diff
@@ -56,50 +46,62 @@ Component._buildAccessors(SoundComponent.prototype, _schema);
 
 Object.assign(SoundComponentSystem.prototype, {
     initializeComponentData: function (component, data, properties) {
-        properties = ['volume', 'pitch', 'positional', 'refDistance', 'maxDistance', 'rollOffFactor', 'distanceModel', 'slots', 'enabled'];
-        ComponentSystem.prototype.initializeComponentData.call(this, component, data, properties);
+        properties = [
+            'distanceModel',
+            'maxDistance',
+            'pitch',
+            'positional',
+            'refDistance',
+            'rollOffFactor',
+            'slots',
+            'volume'
+        ];
+
+        for (var i = 0; i < properties.length; i++) {
+            if (data.hasOwnProperty(properties[i])) {
+                component[properties[i]] = data[properties[i]];
+            }
+        }
+
+        ComponentSystem.prototype.initializeComponentData.call(this, component, data, ['enabled']);
     },
 
     cloneComponent: function (entity, clone) {
-        var key;
-        var oldData = entity.sound.data;
-        var newData = {};
-
-        // copy old data to new data
-        for (key in oldData) {
-            if (oldData.hasOwnProperty(key)) {
-                newData[key] = oldData[key];
-            }
-        }
+        var srcComponent = entity.sound;
+        var srcSlots = srcComponent.slots;
 
         // convert 'slots' back to
         // simple option objects
-        newData.slots = {};
-
-        for (key in oldData.slots) {
-            var oldSlot = oldData.slots[key];
-            if (oldSlot instanceof SoundSlot) {
```

Comment:
Hey @vkalpias , I didn't see how a component being cloned could have a JSON data block instead of an actual `SoundSlot` so I removed that case. Is that OK?

Reply:
that's OK I think